etna/transforms/sklearn.py

```diff
@@ -99,10 +131,31 @@ def inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
         transformed DataFrame.
         """
         if self.inplace:
-            x = df.loc[:, pd.IndexSlice[:, self.in_column]].values
-            transformed = self.transformer.inverse_transform(X=x)
+            if self.mode == TransformMode.per_segment:
+                x = df.loc[:, pd.IndexSlice[:, self.in_column]].values
+                transformed = self.transformer.inverse_transform(X=x)
+
+            elif self.mode == TransformMode.macro:
+                x = self._reshape(df)
+                transformed = self.transformer.inverse_transform(X=x)
+                transformed = self._inverse_reshape(df, transformed)
+
```

Comment:
It's not great finalize without else statement -- this is kind of pattern matching and we should raise Exception or doing something for other possible cases. 
The probablity of self.mode not equal to something from enum is mostly equal to zero. But it can be, for example you change mode by hand in tests and you could pass everything. 
So we should raise ValueError in `else` block

Reply:
Done